src/app/reporting/ReportSchedulerImpl.h

```diff
@@ -62,7 +62,8 @@ class ReportSchedulerImpl : public ReportScheduler
 public:
     using Timeout = System::Clock::Timeout;
 
-    ReportSchedulerImpl(TimerDelegate * aTimerDelegate);
+    ReportSchedulerImpl() = default;
+    ReportSchedulerImpl(TimerDelegate * aTimerDelegate) : ReportScheduler(aTimerDelegate) {}
```

Comment:
Why remove the nullptr check for this constructor?

Reply:
How and where would the Init method get called is the constructor using a TimerDelegate is used?

So far, it doesn't seem it happens. Also, if the constructor accepts a TimerDelegate, how is the Init method supposed to be used?

Reply:
The Init() method is only used right above your other comment (https://github.com/project-chip/connectedhomeip/pull/34316/#discussion_r1696911888) and only for the sReportScheduler in CommonCaseDeviceServerInitParams, which is always default constructed.

Reply:
Which is why the nullptr check should remain in this constructor right? Otherwise the other places in the code using something else than the default constructor just lost their nullptr check.